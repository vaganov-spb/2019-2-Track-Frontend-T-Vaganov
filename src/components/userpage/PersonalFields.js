import React from 'react';
import { connect } from 'react-redux';
import { userInfoSuccess } from '../../actions/index';
import infoStyles from '../../styles/ProfileInfo.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function PersonalFields(props) {
	function onKeyPress(event) {
		if (event.key === 'Enter') {
			event.preventDefault();
		}
	}

	function onChange(event, name) {
		props.changeAttr({[name]: event.target.value });
	}

	if (!props.isLoaded) {
		return null;
	}

	return (
		<div className={infoStyles.wrapper}>
			<div className={infoStyles.field_name}> {props.name} </div>
			<textarea 
				className={infoStyles.name} 
				onKeyPress={onKeyPress} 
				onChange={(e) => onChange(e, props.name)} 
				placeholder={props.placeholder} 
				minLength="5" 
				maxLength={props.maxlen}
				rows={props.row}
				value={props.user[props.name]}
			/>
			<div className={infoStyles.userlen}> {props.description} </div>
		</div>
	);
}

const mapStateToProps = (state) => ({
	user: state.userinfo.info,
	error: state.userinfo.error,
	isLoaded: state.userinfo.isLoaded,
});

const mapDispatchToProps = (dispatch) => ({
	changeAttr: attrObj => dispatch(userInfoSuccess(attrObj))
});

export default connect(mapStateToProps, mapDispatchToProps)(PersonalFields);