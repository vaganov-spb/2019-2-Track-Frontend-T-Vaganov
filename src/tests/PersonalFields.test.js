/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import PersonalFields from '../components/userpage/UserInfoFields';

const mockStore = configureStore();

describe('PersonalFields renders and act correctly', () => {
	let store;
	let component;
	let props;

	beforeEach(() => {
		props = {
			name: 'Name',
			placeholder: '',
			maxlen: 5,
			row: 1,
			description: '',
		};

		store = mockStore({
			userinfo: {
				info: {
					Name: 'Borat',
					Nick: 'Sagdiev',
					Bio: 'Pasha Technik'
				},
				isLoaded: true,
				error: null, 
			},
		});
        
		store.dispatch = jest.fn();

		component = renderer.create(
			<Provider store={store}>
				<MemoryRouter>
					<PersonalFields { ...props }/>
				</MemoryRouter>  
			</Provider>
		);
	});

	it ('should render normally', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
   
	it ('should work correctly onChange', () => {
		const value = component.root.findByProps({className: 'name'});
		const event = {
			preventDefault() {},
			target: { value: '' }
		};

		value.props.onChange(event);
		expect(store.dispatch).toHaveBeenCalledTimes(1);
	});
});