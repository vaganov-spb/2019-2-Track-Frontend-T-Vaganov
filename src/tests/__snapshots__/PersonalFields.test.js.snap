// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalFields renders and act correctly should render normally 1`] = `
<div
  className="wrapper"
>
  <div
    className="field_name"
  >
     
    Name
     
  </div>
  <textarea
    className="name"
    maxLength={5}
    minLength="5"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    rows={1}
    value="Borat"
  />
  <div
    className="userlen"
  >
     
    
     
  </div>
</div>
`;
